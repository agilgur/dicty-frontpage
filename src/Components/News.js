// @flow
import React from "react"
import styled from "styled-components"
import { Flex, Box } from "grid-styled"
import twitterLogo from "./images/twitterLogo.png"
import Octicon from "react-octicon"
import renderHTML from "react-render-html"

const Container = styled.div`
  text-align: left;
  padding-left: 10px;
  padding-right: 10px;
  background-color: #f2f2f2;
  border-radius: 15px;
  margin-top: 10px;
  margin-bottom: 10px;
  margin-left: 1px;
`

const Header = styled.div`
  color: black;
  font-size: 20px;
  padding: 15px 30px 10px 30px;
  vertical-align: top;
  text-align: right;

  @media (max-width: 767px) {
    font-size: 24px;
    text-align: right;
    padding: 20px 5px 20px 15px;
  }
`

const Img = styled.img`width: 30px;`

const NewsBox = styled.ul`
  padding: 0px 25px 10px 0px;
  font-size: 12px;
  margin-bottom: 5px;

  @media (max-width: 992px) and (min-width: 767px) {
    font-size: 10px;
  }
  @media (max-width: 767px) {
    font-size: 16px;
  }
`

const ListItems = styled.li`
  list-style-type: none;
  margin-bottom: 5px;
`

const NewsMore = styled.div`
  color: #0b3861;
  font-size: 11px;
  font-style: italic;
  font-weight: normal;
  text-align: center;
  padding-top: 5px;
  padding-bottom: 15px;
`

const NewsDate = styled.span`
  color: #0b3861;
  padding-right: 10px;
`

const NewsContent = styled.span`padding-right: 10px;`
const NewsSource = styled.span`color: #0b3861;`
const NewsTitle = styled.span`padding-left: 5px;`

const News = (props: { posts: Array<any> }) => {
  const text = props.posts.map(post =>
    <ListItems key={post.id}>
      <NewsDate>
        {post.date}
      </NewsDate>
      <NewsContent>
        {renderHTML(post.content)}
      </NewsContent>
      <NewsSource>
        <strong>Source:</strong>&nbsp;
        {post.source}
      </NewsSource>
    </ListItems>,
  )

  return (
    <Container>
      <Header>
        <Flex wrap>
          <Box px={2} py={1} width={1 / 10}>
            <a
              href="https://twitter.com/dictybase"
              alt="Dicty News at Twitter"
              target="new">
              <Img src={twitterLogo} />
            </a>
          </Box>
          <Box px={2} py={1} width={9 / 10}>
            <Octicon name="globe" mega />
            <NewsTitle>DICTY NEWS</NewsTitle>
          </Box>
        </Flex>
      </Header>
      <NewsBox>
        <ul>
          {text}
        </ul>
      </NewsBox>
      <NewsMore>
        <Octicon name="plus" /> &nbsp;
        <a href="" alt="more news">
          more news{" "}
        </a>
      </NewsMore>
    </Container>
  )
}

export default News
