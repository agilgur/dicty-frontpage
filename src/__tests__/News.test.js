// @flow
import React from 'react';
import { shallow } from 'enzyme';
import News from '../Components/News';
import styled from 'styled-components';
import twitterLogo from '../Components/images/twitterLogo.png';


describe ('<News posts = {posts} />', () => {
  const posts = [
    {
      id: 1,
      date: '2015.08.19',
      source: 'Dicty Stock Center',
      content:
        "<strong>Why axenic strains grow by fluid uptake</strong>: We have added the <a href='http://dictybase.org/gene/DDB_G0350652' target='new'> axeB gene </a> to Chromosome 3 at the locus which is largely deleted in axenic strains and thus responsible for their ability to grow axenically. Only an axeB 3' fragment could be generated on the AX4 genome assembly. The full length gene in non-axenic strains such as DdB codes for the conserved neurofibromin 1, a RasGAP, which has been shown to be involved in macropinocytosis and phagocytosis. For more information, see <a href='http://dictybase.org/publication/pubmed/25815683' target='new'>Bloomfield et al</a>."
    },
    {
      id: 2,
      date: '2015.08.12',
      source: 'National Geographics',
      content:
        "Dicty in the News: The National Geographics series a <a href='http://phenomena.nationalgeographic.com/2015/08/24/the-bacteria-that-turn-amoebas-into-farmers/?sf12476278=1' target='new'>Not Exactly Rocket Science</a> has published a nice piece about the latest results reporting how Burkholderia turns Dicty amooeba into farmers, <a href='http://www.pnas.org/content/early/2015/08/18/1511878112' target='new'>published</a> by Suzanne di Salvo and colleagues from the Strassmann/Queller lab."
    }
  ];

  it('contatins NewsTitle', () => {
    const NewsTitle= "DICTY NEWS";
    const container = shallow(<News posts={posts} />);
    const detail = NewsTitle;
    expect(container.contains(detail)).toEqual(true);
  });

  it('contains 2 <li> items', () => {
    const container = shallow(<News posts={posts} />);
    expect(container.find('li').length).toEqual(2);
  });

  it('contatins 2 <a> tags', () => {
    const container = shallow(<News posts={posts} />);
    expect(container.find('a').length).toEqual(2);
  });

});
