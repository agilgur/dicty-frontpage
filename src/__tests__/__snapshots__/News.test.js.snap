// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News snapshot test 1`] = `
<div
  className="sc-bxivhb cAoBMP"
>
  <div
    className="sc-ifAKCX gnZHCx"
  >
    <div
      className="sc-htpNat hLJahE sc-bdVaJa iHZvIS"
      wrap={true}
    >
      <div
        className="sc-bdVaJa hZXWJA"
      >
        <a
          alt="Dicty News at Twitter"
          href="https://twitter.com/dictybase"
          target="new"
        >
          <img
            className="sc-EHOje bThSEj"
            src="twitterLogo.png"
          />
        </a>
      </div>
      <div
        className="sc-bdVaJa iFPAFX"
      >
        DICTY NEWS
      </div>
    </div>
  </div>
  <ul
    className="sc-gzVnrw dPAoYG"
  >
    <ul>
      <li
        className="sc-htoDjs ikXCMu"
      >
        <div
          className="sc-iwsKbI fSYNJk"
        >
          2015.08.19
        </div>
        <div
          className="sc-gqjmRU cBdDLr"
        >
          Dicty Stock Center
        </div>
        <div
          className="sc-gZMcBi ciewAv"
        >
          <type>
            &lt;strong&gt;Why axenic strains grow by fluid uptake&lt;/strong&gt;: We have added the &lt;a href='http://dictybase.org/gene/DDB_G0350652' target='new'&gt; axeB gene &lt;/a&gt; to Chromosome 3 at the locus which is largely deleted in axenic strains and thus responsible for their ability to grow axenically. Only an axeB 3' fragment could be generated on the AX4 genome assembly. The full length gene in non-axenic strains such as DdB codes for the conserved neurofibromin 1, a RasGAP, which has been shown to be involved in macropinocytosis and phagocytosis. For more information, see &lt;a href='http://dictybase.org/publication/pubmed/25815683' target='new'&gt;Bloomfield et al&lt;/a&gt;.
          </type>
        </div>
      </li>
      <li
        className="sc-htoDjs ikXCMu"
      >
        <div
          className="sc-iwsKbI fSYNJk"
        >
          2015.08.12
        </div>
        <div
          className="sc-gqjmRU cBdDLr"
        >
          National Geographics
        </div>
        <div
          className="sc-gZMcBi ciewAv"
        >
          <type>
            Dicty in the News: The National Geographics series a &lt;a href='http://phenomena.nationalgeographic.com/2015/08/24/the-bacteria-that-turn-amoebas-into-farmers/?sf12476278=1' target='new'&gt;Not Exactly Rocket Science&lt;/a&gt; has published a nice piece about the latest results reporting how Burkholderia turns Dicty amooeba into farmers, &lt;a href='http://www.pnas.org/content/early/2015/08/18/1511878112' target='new'&gt;published&lt;/a&gt; by Suzanne di Salvo and colleagues from the Strassmann/Queller lab.
          </type>
        </div>
      </li>
    </ul>
  </ul>
  <div
    className="sc-dnqmqq hwhtnP"
  >
    <a
      alt="more news"
      href=""
    >
       
      more news
       
    </a>
  </div>
</div>
`;
